store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9129,22 +9136,28 @@ public void deleteFileSharedWithMe(Mailbox mbox, OperationContext octxt, String
     }
 
     public void updateFileSharedWithMe(Mailbox mbox, OperationContext octxt, String ownerAccountID, int ownerFileId,
-            String remoteUuid, String fileOwnerName, String contentType, String rights, String granteeAccountID)
-            throws ServiceException {
+            String remoteUuid, String fileOwnerName, String contentType, String rights, String granteeAccountID,
+            String fileName, Long date, Long size) throws ServiceException {
         CreateFileSharedWithMe redoRecorder = new CreateFileSharedWithMe(mId, remoteUuid);
         boolean success = false;
         try {
             beginTransaction("updateFileSharedWithMe", octxt, redoRecorder);
             Document item = (Document) mbox.getItemByUuid(remoteUuid + ":" + ownerFileId, Type.DOCUMENT);
             if (item.getOwnerAccountId().equals(ownerAccountID) && item.getAccountId().equals(granteeAccountID)) {
                 Metadata meta = new Metadata();
-                meta.put(Metadata.FN_CREATOR, fileOwnerName);
-                meta.put(Metadata.FN_MIME_TYPE, contentType);
-                meta.put(Metadata.FN_FILE_PERMISSION, rights);
+                meta.put(Metadata.FN_CREATOR, fileOwnerName==""?item.creator:fileOwnerName);
+                meta.put(Metadata.FN_MIME_TYPE, contentType==""?item.contentType:contentType);
+                meta.put(Metadata.FN_FILE_PERMISSION, rights==""?item.permission:rights);
                 meta.put(Metadata.FN_OWNER_ACCOUNT_ID, ownerAccountID);
                 meta.put(Metadata.FN_OWNER_FILE_ID, ownerFileId);
                 mbox.getItemCache().remove(item.getId());
-                new DbMailItem(mbox).update(item, meta);
+                // rename operation, document revision
+                if (fileName != "" && fileName != null && date != -1) {
```

Comment:
can use StringUtil.isNullOrEmpty for filename

Reply:
Added check for fileName as suggested
for date, -1 is the default value. hence this scenario would fail in case of file permission modification and would go to else block as required and updates permission through the else block update method